### PR TITLE
[RLlib] Better docstrings and warnings and granularity for space checks in MultiAgentEnv

### LINKED ISSUE #28706: [RLlib] Add documentation for "preferred format" for observation space and action space
### Description

When defining a multi-agent environment, you can specify observation spaces and action spaces in the new "preferred format". There is a check for this and a possible warning. But I couldn't find documentation anywhere about this "preferred format" and how to correctly specify an observation space.

### Link

_No response_

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
## Why are these changes needed?

Our API currently gives little hints on what to do to properly deal with spaces under the MultiAgentEnv API.

## Related issue number

Closes #28706 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by sven1977 — APPROVED** [review]
Cool, a very nice PR! Thanks for this enhancement @ArturNiederfahrenhorst .

**Comment by sven1977** [comment]
Kicked off the 2 failed tests. Otherwise, good to merge.

File: rllib/env/multi_agent_env.py [hunk 1]
```diff
@@ -33,6 +33,11 @@ class MultiAgentEnv(gym.Env):
     Agents are identified by (string) agent ids. Note that these "agents" here
     are not to be confused with RLlib Algorithms, which are also sometimes
     referred to as "agents" or "RL agents".
+
+    The preferred format for action- and observation space is a mapping from agent
```
Comment (cool-RR) [inline rllib/env/multi_agent_env.py:37]: "The preferred format for action- and observation space is a mapping" Any kind of mapping? Like a dict?

Comment (ArturNiederfahrenhorst) [reply rllib/env/multi_agent_env.py:37]: @cool-RR Yes, a dict!

Comment (cool-RR) [reply rllib/env/multi_agent_env.py:37]: It should probably say dict then.

**Pull request merged**